plugins/ingest/src/main/java/org/elasticsearch/plugin/ingest/transport/simulate/SimulateExecutionService.java

```diff
@@ -43,7 +44,7 @@ SimulateDocumentResult executeDocument(Pipeline pipeline, IngestDocument ingestD
             List<SimulateProcessorResult> processorResultList = new ArrayList<>();
             IngestDocument currentIngestDocument = new IngestDocument(ingestDocument);
             for (int i = 0; i < pipeline.getProcessors().size(); i++) {
-                Processor processor = pipeline.getProcessors().get(i);
+                HandledProcessor processor = pipeline.getProcessors().get(i);
```

Comment:
I think that this is a significant change to the simulate api? maybe that should be done in a different  pr?